pkg/controller/secretannotator/azure/reconciler_test.go

```diff
@@ -47,37 +129,46 @@ func TestAzureSecretAnnotatorReconcile(t *testing.T) {
 			ccazure.AzureSubscriptionID: []byte("AZURE_SUBSCRIPTION_ID"),
 			ccazure.AzureTenantID:       []byte("AZURE_TENANT_ID"),
 		},
-	}}
+	}
 
-	fakeClient := fake.NewFakeClient(existingSecret...)
+}
 
-	rcc := &ReconcileCloudCredSecret{
-		Client: fakeClient,
-		Logger: log.WithField("controller", "testController"),
-	}
+func setupAdalMock(r *mock.MockAdalClientMockRecorder, roles []string) {
+	gomock.InOrder(
+		// these methdods are extensivly tested in azure codebase
+		r.NewOAuthConfig(gomock.Any(), gomock.Any()).Return(&TestOAuthConfig, nil),
+		r.NewServicePrincipalToken(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(newServicePrincipalTokenManual(roles), nil),
```

Comment:
similar comment here. `gomock.Eq(TestOAuthConfig), gomock.Eq(TestClientID), gomock.Eq(TestClientSecret), gomock.Eq(azure.PublicCloud.GraphEndpoint)`
(with the necessary updates to the generated secret in getInputSecret() )